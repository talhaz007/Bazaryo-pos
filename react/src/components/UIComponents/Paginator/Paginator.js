import React from "react";
import ReactPaginate from "react-paginate";

import "./Paginator.scss";

const Paginator = (props) => {
  const {
    className,
    pageCount,
    pageRangeDisplayed,
    marginPagesDisplayed,
    onPageChange,
  } = props;
  const buttonImg = <div className="page-button-controller" />;
  return (
    <div className={`paginator-wrapper ${className || ""}`}>
      <ReactPaginate
        pageCount={pageCount}
        pageRangeDisplayed={pageRangeDisplayed || 4}
        marginPagesDisplayed={marginPagesDisplayed || 1}
        onPageChange={onPageChange}
        previousLabel={buttonImg}
        nextLabel={buttonImg}
        containerClassName="pagination"
        pageClassName="page"
        activeClassName="active-page"
        previousClassName="prev-page controller-page"
        nextClassName="next-page controller-page"
      />
    </div>
  );
};

export default Paginator;
