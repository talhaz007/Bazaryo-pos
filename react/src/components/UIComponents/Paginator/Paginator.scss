.paginator-wrapper {
    max-width: 500px;
    .pagination {
        display: flex;
        padding: 0;
        margin: 21px 0;
        list-style: none;
        outline: none;
        align-items: center;

        .page {
            border-radius: 4px;
            a {
                padding: 4px 11px;
                text-decoration: none;
                line-height: 24px;
                color: #6f402f;
                &:focus {
                    outline: none;
                }
            }
            &:active {
                background: #6f402f;
                a {
                    color: #fff;
                    line-height: 24px;
                }
            }
            &.active-page {
                background: #6f402f;
                a {
                    color: #fff;
                    line-height: 24px;
                }
            }
        }
        .break {
            a {
                padding: 4px 11px;
                padding: 4px 11px;
                text-decoration: none;
                line-height: 24px;
                color: #6f402f;
            }
        }
        .controller-page {
            display: flex;
            a {
                .page-button-controller {
                    background-image: url("../../../Assets/img/slider-controller-active.svg");
                    background-position: center;
                    background-repeat: no-repeat;
                    background-size: cover;
                    width: 7.41px;
                    height: 12px;
                }
                &:focus {
                    outline: none;
                }
            }
            &:active {
                a {
                    .page-button-controller {
                        background-image: url("../../../Assets/img/slider-controller-active.svg");
                    }
                }
            }
        }
        .next-page {
            margin-left: 37px;
        }
        .prev-page {
            margin-right: 37px;
            .page-button-controller {
                transform: rotate(-180deg);
            }
        }
        a {
            padding: 4px 11px;
            text-decoration: none;
            line-height: 24px;
            color: #8b90a0;
        }
    }
}
