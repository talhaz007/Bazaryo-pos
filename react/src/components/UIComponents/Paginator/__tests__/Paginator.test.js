import React from 'react';
import {shallow} from 'enzyme';
import {describe, it, expect} from '@jest/globals';
import Paginator from '../Paginator';

describe('Paginator', () => {
    let wrapper;

    it('should be rendered', () => {
        wrapper = shallow(<Paginator/>);
        expect(wrapper).toMatchSnapshot();
    });
});
