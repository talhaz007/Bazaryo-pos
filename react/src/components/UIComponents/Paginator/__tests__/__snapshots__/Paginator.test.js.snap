// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should be rendered 1`] = `
<div
  className="paginator-wrapper "
>
  <o
    activeClassName="active-page"
    breakLabel="..."
    containerClassName="pagination"
    disableInitialCallback={false}
    disabledClassName="disabled"
    eventListener="onClick"
    marginPagesDisplayed={1}
    nextAriaLabel="Next page"
    nextClassName="next-page controller-page"
    nextLabel={
      <div
        className="page-button-controller"
      />
    }
    nextRel="next"
    pageClassName="page"
    pageCount={10}
    pageRangeDisplayed={4}
    prevRel="prev"
    previousAriaLabel="Previous page"
    previousClassName="prev-page controller-page"
    previousLabel={
      <div
        className="page-button-controller"
      />
    }
  />
</div>
`;
